grr/server/grr_response_server/gui/api_integration_tests/metadata_test.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+# Lint as: python3
+# -*- encoding: utf-8 -*-
+"""Tests for API client and metadata-related API calls."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+import pkg_resources
+
+from absl import app
+
+from grr_response_server.gui import api_integration_test_lib
+from grr.test_lib import skip
+from grr.test_lib import test_lib
+
+
+class ApiClientLibMetadataTest(api_integration_test_lib.ApiIntegrationTest):
+  """Tests metadata-related part of GRR Python API client library."""
+
+  @skip.If(
+    "openapi-spec-validator" not in {p.key for p in pkg_resources.working_set},
+    "The openapi-spec-validator module used for validating the OpenAPI "
+    "specification is not installed."
+  )
+  def testOpenApiDescriptionValidity(self):
+    import openapi_spec_validator
+    openapi_dict = self.api.GetOpenApiDescription(as_string=False)
+    try:
+      errors = openapi_spec_validator.validate_spec(openapi_dict)
+    except:
+      errors = True
+
+    self.assertIsNone(errors)
```

Comment:
I wouldn't put the validator call into try/except. Currently, if the test fails, it's not possible to distinguish cases when it reported errors or actually raised.

Reply:
My understanding is that we can treat this is as a default assumption: the function that we're testing shouldn't raise. If it should raise, though, then we mark the code with self.assertRaises and the expectation of the test becomes clear.